### PR TITLE
Components that aren't axes/plots removing instance methods from constructor

### PR DESCRIPTION
**Pull request by bluong**

### CODE REVIEW

File: src/components/abstractComponent.ts [hunk 1]
```diff
@@ -274,18 +274,29 @@ export module Component {
...
+
+    private static ensureXAlignment(alignment: string): string {
+      alignment = alignment.toLowerCase();
+      if (["left", "center", "right"].indexOf(alignment) === -1) {
+        throw new Error("Unsupported alignment");
+      }
+      return alignment;
+    }
+
+    public static _xAlignmentToProportion(alignment: string): number {
```
Comment (jtlan) [inline src/components/abstractComponent.ts]: public _? Could this be `protected`?

File: src/components/abstractComponent.ts [hunk 2]
```diff
@@ -300,18 +311,29 @@ export module Component {
...
+
+    private static ensureYAlignment(alignment: string): string {
+      alignment = alignment.toLowerCase();
+      if (["top", "center", "bottom"].indexOf(alignment) === -1) {
+        throw new Error("Unsupported alignment");
+      }
+      return alignment;
+    }
+
+    public static _yAlignmentToProportion(alignment: string): number {
```
Comment (jtlan) [inline src/components/abstractComponent.ts]: Same comment as for `_xAlignmentToProportion`: Could this be made `protected`?

File: src/components/gridlines.ts [hunk 3]
```diff
@@ -23,15 +23,15 @@ export module Component {
...
-      }
-      if (this._yScale) {
-        this._yScale.broadcaster.registerListener(this, () => this._render());
-      }
+    }
+
+    public _anchor(element: D3.Selection) {
+      super._anchor(element);
+      this._xScale.broadcaster.registerListener(this, () => this._render());
+      this._yScale.broadcaster.registerListener(this, () => this._render());
```
Comment (jtlan) [inline src/components/gridlines.ts]: This is going to cause a bug if someone passes in `null` for either scale.

File: src/utils/utils.ts [hunk 4]
```diff
@@ -344,6 +344,12 @@ export module _Util {
 
       return "#" + rHex + gHex + bHex;
     }
+
+    export function uniqAdd<T>(arr: T[], item: T) {
```
Comment (jtlan) [inline src/utils/utils.ts]: "`uniqPush`"?

File: src/components/abstractComponent.ts (hunk 1 above)
Comment (bluong) [inline src/components/abstractComponent.ts]: I don't think there's such thing as protected static... I'll check though

File: src/components/gridlines.ts (hunk 3 above)
Comment (bluong) [inline src/components/gridlines.ts]: But the constructor I think specifies that one must exist i think...?  If not I'll change it

File: src/utils/utils.ts (hunk 4 above)
Comment (bluong) [inline src/utils/utils.ts]: Probably better

File: src/components/abstractComponent.ts (hunk 1 above)
Comment (bluong) [inline src/components/abstractComponent.ts]: Welp apparnetly compiler is ok with this.  Will change

**Pull request merged**